src/basic/build-path.c

```diff
@@ -33,7 +33,14 @@ static int get_runpath_from_dynamic(const ElfW(Dyn) *d, const char **ret) {
                         break;
 
                 case DT_STRTAB:
-                        strtab = (const char *) d->d_un.d_val;
+                        /* On MIPS and RISC-V DT_STRTAB records an offset, not a valid address, so it has to be adjusted
+                         * using the bias calculated earlier. */
+                        if ((const char *) d->d_un.d_val)
+                                strtab = (const char *) ((size_t) d->d_un.d_val
+#if defined(__mips__) || defined(__riscv)
```

Comment:
Maybe, uintptr_t?

Reply:
hmm, i have no idea if size_t or uintptr_t are the better type here.